tests/referral_tokens/test_exchange_referral_token.py

```diff
@@ -0,0 +1,179 @@
+import pytest
+from graphql_relay import to_global_id
+from creator.studies.models import Study
+from django.contrib.auth.models import Group
+from creator.referral_tokens.models import ReferralToken
+from creator.studies.factories import StudyFactory
+from creator.referral_tokens.factories import ReferralTokenFactory
+
+EXCHANGE_REFERRALTOKEN = """
+mutation ($token: ID!) {
+  exchangeReferralToken(token: $token) {
+    message
+    user {
+      email
+      studies {
+        edges {
+          node {
+            id
+          }
+        }
+      }
+      groups {
+        edges {
+          node {
+            id
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+CREATE_REFERRALTOKEN = """
+mutation newReferralToken($input: ReferralTokenInput!) {
+  createReferralToken(input: $input) {
+    referralToken {
+      uuid
+    }
+  }
+}
+"""
+
+
+@pytest.mark.parametrize(
+    "user_group,allowed",
+    [
+        ("Administrators", True),
+        ("Services", True),
+        ("Developers", True),
+        ("Investigators", True),
+        ("Bioinformatics", True),
+        (None, False),
+    ],
+)
+def test_exchange_referral_token(db, clients, user_group, allowed):
+    """
+    All login user can exchange referral token
```

Comment:
To further test completeness:
- Create new study with `StudyFactory`
- Add new study to the token
- Check that the user is not a collaborator on the study
- Run the query
- Check that the user is a collaborator on the study